### PR TITLE
Paligemma - fix slow tests, add bf16 and f16 slow tests

### PR DESCRIPTION
**Pull request by molbap**
# What does this PR do?

As per PR title.
cc @ydshieh

### CODE REVIEW

File: tests/models/paligemma/test_modeling_paligemma.py [hunk 1]
```diff
@@ -272,48 +272,48 @@ def tearDown(self):
...
-         256000, 256000, 256000, 256000, 256000, 256000, 256000, 256000, 256000,
-         256000, 256000, 256000, 256000, 256000, 256000, 256000, 256000, 256000,
-         256000, 256000, 256000, 256000, 256000, 256000, 256000, 256000, 256000,
-         256000, 256000, 256000, 256000, 256000, 256000, 256000, 256000, 256000,
-         256000, 256000, 256000, 256000, 256000, 256000, 256000, 256000, 256000,
-         256000, 256000, 256000, 256000, 256000, 256000, 256000, 256000, 256000,
-         256000, 256000, 256000, 256000, 256000, 256000, 256000, 256000, 256000,
-         256000, 256000, 256000, 256000, 256000, 256000, 256000, 256000, 256000,
-         256000, 256000, 256000, 256000,      2,    108]])
+        EXPECTED_INPUT_IDS = torch.tensor([[257152, 257152, 257152, 257152, 257152, 257152, 257152, 257152, 257152,
```
Comment (ydshieh) [inline tests/models/paligemma/test_modeling_paligemma.py:283]: could we use [257152] * num + [... whatever else] ?

File: tests/models/paligemma/test_modeling_paligemma.py [hunk 2]
```diff
@@ -322,11 +322,11 @@ def test_small_model_integration_test(self):
...
     @slow
     @require_bitsandbytes
-    def test_small_model_integration_test_paligemma(self):
+    def test_small_model_integration_test_paligemma_VQA(self):
         # Let' s make sure we test the preprocessing to replace what is used
-        model_id = "gv-hf/PaliGemma-test-224px-hf"
+        model_id = "google/paligemma-3b-pt-224"
 
-        model = PaliGemmaForConditionalGeneration.from_pretrained("gv-hf/PaliGemma-test-224px-hf")
+        model = PaliGemmaForConditionalGeneration.from_pretrained("google/paligemma-3b-pt-224")
```
Comment (ydshieh) [inline tests/models/paligemma/test_modeling_paligemma.py:329]: this is just model_id ?

File: tests/models/paligemma/test_modeling_paligemma.py [hunk 3]
```diff
@@ -344,11 +344,35 @@ def test_small_model_integration_test_paligemma(self):
             EXPECTED_DECODED_TEXT,
         )
 
+    @slow
+    @require_bitsandbytes
+    def test_small_model_integration_test_paligemma_empty_prompt(self):
+        # Let' s make sure we test the preprocessing to replace what is used
+        model_id = "google/paligemma-3b-pt-224"
+
+        model = PaliGemmaForConditionalGeneration.from_pretrained("google/paligemma-3b-pt-224")
```
Comment (ydshieh) [inline tests/models/paligemma/test_modeling_paligemma.py:353]: same comment

File: tests/models/paligemma/test_modeling_paligemma.py [hunk 4]
```diff
@@ -322,11 +322,11 @@ def test_small_model_integration_test(self):
 
     @slow
     @require_bitsandbytes
```
Comment (ydshieh) [inline tests/models/paligemma/test_modeling_paligemma.py:324]: same comment

File: tests/models/paligemma/test_modeling_paligemma.py [hunk 5]
```diff
@@ -369,15 +393,49 @@ def test_small_model_integration_test_paligemma_batched(self):
...
         output = model.generate(**inputs, max_new_tokens=20)
 
-        EXPECTED_DECODED_TEXT = ["answer en Where is the cow standing?\nbeach", "\ncow standing on the beach"]  # fmt: skip
+        EXPECTED_DECODED_TEXT = ["answer en Where is the cow standing?\nbeach", "\ncow on the beach"]  # fmt: skip
 
         self.assertEqual(processor.batch_decode(output, skip_special_tokens=True), EXPECTED_DECODED_TEXT)
 
     @slow
+    @require_torch
+    @require_bitsandbytes
```
Comment (ydshieh) [inline tests/models/paligemma/test_modeling_paligemma.py:402]: same

File: tests/models/paligemma/test_modeling_paligemma.py [hunk 6]
```diff
@@ -369,15 +393,49 @@ def test_small_model_integration_test_paligemma_batched(self):
...
+            .to(torch_device)
+        )
+        output = model.generate(**inputs, max_new_tokens=20)
+
+        EXPECTED_DECODED_TEXT = ["answer en Where is the cow standing?\nbeach", "\ncow on the beach"]  # fmt: skip
+        self.assertEqual(self.processor.batch_decode(output, skip_special_tokens=True), EXPECTED_DECODED_TEXT)
+
+    @slow
+    @require_torch
     @require_bitsandbytes
```
Comment (ydshieh) [inline tests/models/paligemma/test_modeling_paligemma.py:433]: same

**Review by ydshieh — COMMENTED** [review]
Thanks @molbap . Some nits.

Let's try the `run-slow` feature to make sure everything is passing on CI runner.

**Comment by younesbelkada** [comment]
@molbap for that you can use `require_read_token` context manager: https://github.com/huggingface/transformers/blob/977ce58a78fc46f89151944da745188d2f21332e/tests/models/llama/test_modeling_llama.py#L475 :D !

**Comment by molbap** [comment]
cc @ydshieh should be good to go!

**Review by ydshieh — APPROVED** [review]
Thank you @molbap ! LGTM.

I am not sure I understand the `to replace what is used` part in this comment

> Let' s make sure we test the preprocessing to replace what is used

(what is `what`  and what is `used` here).

I mean it might be good to be more precise - but maybe I lack some context

**Comment by molbap** [comment]
Apparently it comes from Llava code - I have no idea what this comment means!

**Comment by ydshieh** [comment]
In this case, we can merge, and once @younesbelkada has answered, we can update the comments in another PR (if needed)

**Review by younesbelkada — APPROVED** [review]
Thanks !
About your question, honestly I don't know, I think we can safely ignore it

**Comment by ydshieh** [comment]
If that comment is that confusing, let's just remove it in another PR (both here and in LLava testing file). WDTY?

**Review by ArthurZucker — APPROVED** [review]

**Pull request merged**